core/src/main/java/org/elasticsearch/rest/action/search/RestSearchAction.java

```diff
@@ -130,15 +130,12 @@ private static void parseSearchSource(final SearchSourceBuilder searchSourceBuil
             searchSourceBuilder.query(queryBuilder);
         }
 
-        int from = request.paramAsInt("from", -1);
-        if (from != -1) {
-            searchSourceBuilder.from(from);
+        if (request.hasParam("from")) {
```

Comment:
Accepting -1 over the rest layer has never been documented, and it surprised me that it works. I'd be ok with rejecting it